src/dune_rules/lib_rules.ml

```diff
@@ -14,8 +14,9 @@ let msvc_hack_cclibs =
       Option.value ~default:lib (String.drop_prefix ~prefix:"-l" lib))
 
 (* Build an OCaml library. *)
-let build_lib (lib : Library.t) ~native_archives ~sctx ~expander ~flags ~dir
+let build_lib (lib : Library.t) ~native_archives ~cctx ~expander ~flags ~dir
     ~mode ~cm_files =
+  let sctx = Compilation_context.super_context cctx in
```

Comment:
Why do we need the changes in this file?

Reply:
we don't need them, removed in https://github.com/ocaml/dune/pull/6349/commits/a1fc41fe10a5d617ce4dec55971660fc8d694ffe.

as a side note, it's better to hold a bit on any further reviews for now. I am implementing the changes that I suggested in https://github.com/ocaml/dune/pull/6349#discussion_r1010327784, so the PR will go through a few changes.